core/src/test/java/com/novoda/merlin/MerlinsBeardShould.java

```diff
@@ -22,82 +28,149 @@
     @Mock
     private NetworkInfo mockNetworkInfo;
 
+    @Mock
+    private AndroidVersion mockAndroidVersion;
+
     private MerlinsBeard merlinsBeard;
 
     @Before
     public void setUp() {
         initMocks(this);
 
-        merlinsBeard = new MerlinsBeard(mockConnectivityManager);
+        merlinsBeard = new MerlinsBeard(mockConnectivityManager, mockAndroidVersion);
     }
 
     @Test
     public void returnFalseForIsConnectedWhenNetworkConnectionIsUnavailable() {
         when(mockConnectivityManager.getActiveNetworkInfo()).thenReturn(mockNetworkInfo);
         when(mockNetworkInfo.isConnected()).thenReturn(false);
 
-        assertThat(merlinsBeard.isConnected()).isFalse();
+        boolean connected = merlinsBeard.isConnected();
+
+        assertThat(connected).isFalse();
     }
 
     @Test
     public void returnFalseForIsConnectedWhenNetworkConnectionIsNull() {
         when(mockConnectivityManager.getActiveNetworkInfo()).thenReturn(null);
 
-        assertThat(merlinsBeard.isConnected()).isFalse();
+        boolean connected = merlinsBeard.isConnected();
+
+        assertThat(connected).isFalse();
     }
 
     @Test
     public void returnTrueForIsConnectedWhenNetworkConnectionIsAvailable() {
         when(mockConnectivityManager.getActiveNetworkInfo()).thenReturn(mockNetworkInfo);
         when(mockNetworkInfo.isConnected()).thenReturn(true);
 
-        assertThat(merlinsBeard.isConnected()).isTrue();
+        boolean connected = merlinsBeard.isConnected();
+
+        assertThat(connected).isTrue();
     }
 
     @Test
-    public void returnTrueForIsConnectedToWifiWhenNetworkConnectedToWifiIsConnected() {
+    public void returnTrueForIsConnectedToWifiWhenNetworkConnectedToWifiIsConnectedAndAndroidVersionIsBelowMarshmallow() {
```

Comment:
Typo? Maybe `WhenNetworkConnectedToWifiIsConnected` doesn't need the `Network` and `IsConnected`. It's a long test case name, `WhenConnectedToWifi` might be enough information.